From ec35f3ee72bc9610438251494bf7fbf21b291af0 Mon Sep 17 00:00:00 2001
From: MatVeiQaaa <matthehard@gmail.com>
Date: Wed, 22 Apr 2026 12:31:14 +0200
Subject: [PATCH 2/2] remove dllmain

---
 mecab/src/libmecab.cpp | 76 +++++++++++++++++++++---------------------
 1 file changed, 38 insertions(+), 38 deletions(-)

diff --git a/mecab/src/libmecab.cpp b/mecab/src/libmecab.cpp
index 296382b..3bab0e4 100644
--- a/mecab/src/libmecab.cpp
+++ b/mecab/src/libmecab.cpp
@@ -41,44 +41,44 @@ void setGlobalError(const char *str) {
 
 HINSTANCE DllInstance = 0;
 
-extern "C" {
-  BOOL WINAPI DllMain(HINSTANCE hinst, DWORD dwReason, LPVOID) {
-    LPVOID data = 0;
-    if (!DllInstance) {
-      DllInstance = hinst;
-    }
-    switch (dwReason) {
-      case DLL_PROCESS_ATTACH:
-        if ((g_tls_index = ::TlsAlloc()) == TLS_OUT_OF_INDEXES) {
-          return FALSE;
-        }
-        // Not break in order to initialize the TLS.
-      case DLL_THREAD_ATTACH:
-        data = (LPVOID)::LocalAlloc(LPTR, kErrorBufferSize);
-        if (data) {
-          ::TlsSetValue(g_tls_index, data);
-        }
-        break;
-      case DLL_THREAD_DETACH:
-        data = ::TlsGetValue(g_tls_index);
-        if (data) {
-          ::LocalFree((HLOCAL)data);
-        }
-        break;
-      case DLL_PROCESS_DETACH:
-        data = ::TlsGetValue(g_tls_index);
-        if (data) {
-          ::LocalFree((HLOCAL)data);
-        }
-        ::TlsFree(g_tls_index);
-        g_tls_index = TLS_OUT_OF_INDEXES;
-        break;
-      default:
-        break;
-    }
-    return TRUE;
-  }
-}
+//extern "C" {
+//  BOOL WINAPI DllMain(HINSTANCE hinst, DWORD dwReason, LPVOID) {
+//    LPVOID data = 0;
+//    if (!DllInstance) {
+//      DllInstance = hinst;
+//    }
+//    switch (dwReason) {
+//      case DLL_PROCESS_ATTACH:
+//        if ((g_tls_index = ::TlsAlloc()) == TLS_OUT_OF_INDEXES) {
+//          return FALSE;
+//        }
+//        // Not break in order to initialize the TLS.
+//      case DLL_THREAD_ATTACH:
+//        data = (LPVOID)::LocalAlloc(LPTR, kErrorBufferSize);
+//        if (data) {
+//          ::TlsSetValue(g_tls_index, data);
+//        }
+//        break;
+//      case DLL_THREAD_DETACH:
+//        data = ::TlsGetValue(g_tls_index);
+//        if (data) {
+//          ::LocalFree((HLOCAL)data);
+//        }
+//        break;
+//      case DLL_PROCESS_DETACH:
+//        data = ::TlsGetValue(g_tls_index);
+//        if (data) {
+//          ::LocalFree((HLOCAL)data);
+//        }
+//        ::TlsFree(g_tls_index);
+//        g_tls_index = TLS_OUT_OF_INDEXES;
+//        break;
+//      default:
+//        break;
+//    }
+//    return TRUE;
+//  }
+//}
 #else  // _WIN32
 namespace {
 #ifdef HAVE_TLS_KEYWORD
-- 
2.53.0.windows.2

