From 31cc0687223819c0c4e72389c9d56ed489f97d37 Mon Sep 17 00:00:00 2001
From: MatVeiQaaa <matthehard@gmail.com>
Date: Wed, 22 Apr 2026 11:56:49 +0200
Subject: [PATCH 1/2] static link instead of dllimport

---
 mecab/src/mecab.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mecab/src/mecab.h b/mecab/src/mecab.h
index ce7f265..3a40999 100644
--- a/mecab/src/mecab.h
+++ b/mecab/src/mecab.h
@@ -341,7 +341,7 @@ extern "C" {
 #    define MECAB_DLL_EXTERN  __declspec(dllexport)
 #    define MECAB_DLL_CLASS_EXTERN  __declspec(dllexport)
 #  else
-#    define MECAB_DLL_EXTERN  __declspec(dllimport)
+#    define MECAB_DLL_EXTERN
 #  endif
 #endif
 
-- 
2.53.0.windows.2

